src/libstore/lock.cc

```diff
@@ -7,6 +7,34 @@
 
 namespace nix {
 
+#if __linux__
+
+static std::vector<gid_t> get_group_list(const char *username, gid_t group_id)
+{
+    int ngroups {32}; // arbitrary initial guess
+    std::vector<gid_t> gids;
+    gids.resize(ngroups);
+
+    auto getgroupl {[&] {
+        return getgrouplist(username, group_id, gids.data(), &ngroups);
+    }};
```

Comment:
Ah good to combine with what I wrote below.
```suggestion
    auto getgroupl {[&] {
        int ngroups;
        int err = getgrouplist(username, group_id, gids.data(), &ngroups);
        gids.resize(ngroups);
        return err;
    }};
```
What is confusing is how ngroups looks reused from call but call but is not.

Reply:
`ngroups` *is* reused from call to call.
The change that you are suggesting violates this line of the docs: `Up to *ngroups of these groups are returned in the array groups.`